diff -rupN zlib-1.2.11.orig/zconf.h zlib-1.2.11/zconf.h
--- zlib-1.2.11.orig/zconf.h	2017-01-01 02:37:10.000000000 -0500
+++ zlib-1.2.11/zconf.h	2021-09-24 19:57:28.855205500 -0400
@@ -504,10 +504,10 @@ typedef uLong FAR uLongf;
 #  define z_off_t long
 #endif
 
-#if !defined(_WIN32) && defined(Z_LARGE64)
+#if defined(Z_LARGE64)
 #  define z_off64_t off64_t
 #else
-#  if defined(_WIN32) && !defined(__GNUC__) && !defined(Z_SOLO)
+#  if defined(_WIN32) && !defined(Z_SOLO)
 #    define z_off64_t __int64
 #  else
 #    define z_off64_t z_off_t
diff -rupN zlib-1.2.11.orig/zconf.h.cmakein zlib-1.2.11/zconf.h.cmakein
--- zlib-1.2.11.orig/zconf.h.cmakein	2017-01-01 02:37:10.000000000 -0500
+++ zlib-1.2.11/zconf.h.cmakein	2021-09-24 19:57:28.866206900 -0400
@@ -506,10 +506,10 @@ typedef uLong FAR uLongf;
 #  define z_off_t long
 #endif
 
-#if !defined(_WIN32) && defined(Z_LARGE64)
+#if defined(Z_LARGE64)
 #  define z_off64_t off64_t
 #else
-#  if defined(_WIN32) && !defined(__GNUC__) && !defined(Z_SOLO)
+#  if defined(_WIN32) && !defined(Z_SOLO)
 #    define z_off64_t __int64
 #  else
 #    define z_off64_t z_off_t
diff -rupN zlib-1.2.11.orig/zconf.h.in zlib-1.2.11/zconf.h.in
--- zlib-1.2.11.orig/zconf.h.in	2017-01-01 02:37:10.000000000 -0500
+++ zlib-1.2.11/zconf.h.in	2021-09-24 19:57:28.877208300 -0400
@@ -504,10 +504,10 @@ typedef uLong FAR uLongf;
 #  define z_off_t long
 #endif
 
-#if !defined(_WIN32) && defined(Z_LARGE64)
+#if defined(Z_LARGE64)
 #  define z_off64_t off64_t
 #else
-#  if defined(_WIN32) && !defined(__GNUC__) && !defined(Z_SOLO)
+#  if defined(_WIN32) && !defined(Z_SOLO)
 #    define z_off64_t __int64
 #  else
 #    define z_off64_t z_off_t
